builtin/providers/google/provider.go

```diff
@@ -266,3 +267,6 @@ func handleNotFoundError(err error, d *schema.ResourceData, resource string) err
 
 	return fmt.Errorf("Error reading %s: %s", resource, err)
 }
+
+// This is a global MutexKV for use within this plugin.
```

Comment:
Would be cool to explain what it's locking on, and what limitation makes the lock necessary.